### PR TITLE
Rb 4.1.x nadenislamarre x86

### PR DESCRIPTION
**Pull request by nadenislamarre**
x86 first pr.

almost all is working correctly

Some things to continue to work on :
- es : keyboard works but not in textbox
- kodi : keyboard doesn't work if launched from es
- hardware : wiimote patch
- emulator : Scumm (compilation fails) (not analysed)
- emulator : reicast doesn't work
- emulator : zoom is not correct (too zoomed (sega) or not enought zoomed (gameboy))
- acpi : my power button doesnt ask the recalbox to shut down
- emulator : adjust resolution for performances (lower resolution / list of available resolutions)
- review the upgrade process (check it works)

Some things to know before merging :
- possible incompatibilities with noobs for rpi (i don't know noobs, and i changed the way the recalbox find its partition, see recalbox-part.sh)
- there is a dependance between x86 and x86_64 built : the configurable path ../recalbox-buildroot-x86_64/output/images/EFI.tar.gz is required so that 64 bit computer with efi can boot from the usb key.
  I can provide this file externally.
- x86_64 port is already done too. but i wait that x86 is finnished to finnish it.
- please read each comment of each commit
- and don't hesitate if you've any question.

### CODE REVIEW

File: board/recalbox/fsoverlay/etc/fstab [hunk 1]
```diff
@@ -1,12 +1,10 @@
 # /etc/fstab: static file system information.
 #
 # <file system> <mount pt>     <type>	<options>         <dump> <pass>
-/dev/root       /              ext4	ro,noauto         0      1
```
Comment (digitalLumberjack) [inline board/recalbox/fsoverlay/etc/fstab:4]: noobs sed those two variables.
https://github.com/recalbox/recalbox-rescue/blob/rb-4/sdcontent/os/recalboxOS-rpi3/partition_setup.sh#L16

Did not read the whole pr yet though

**Comment by digitalLumberjack** [comment]
I'll keep reading tomorrow :) great job

**Comment by substring** [comment]
Noobs for pc ? Grub, yeah !

**Comment by digitalLumberjack** [comment]
read again :D
I said noobs sed (not set) it, so if it disapear, may cause issues (for multiboot, or even installation :)

**Comment by nadenislamarre** [comment]
@substring note that we can do better than noobs for pc ;-)
if in a future we do efi + squashfs, install recalbox on a usb key is just an unzip on a fat32 key => no need of boot and recalbox partition, only 1. this is not for now.

File: board/recalbox/fsoverlay/etc/init.d/S02splash [hunk 2]
```diff
@@ -51,7 +51,15 @@ do_start ()
             dbus-send --print-reply=literal --session --dest=org.mpris.MediaPlayer2.omxplayer /org/mpris/MediaPlayer2 org.mpris.MediaPlayer2.Player.Action int32:15 >/dev/null
         fi
     fi
-    fbv -f -i /recalbox/system/resources/splash/logo-version.png
+
+    # on some sytems, fb0 is not immediatly loaded, so, keep a chance by waiting a bit
+    N=0
```
Comment (digitalLumberjack) [inline board/recalbox/fsoverlay/etc/init.d/S02splash:56]: N=15

File: board/recalbox/fsoverlay/etc/init.d/S02splash [hunk 3]
```diff
@@ -51,7 +51,15 @@ do_start ()
...
+    while ! test -e /dev/fb0 -o $N -gt 15
```
Comment (digitalLumberjack) [inline board/recalbox/fsoverlay/etc/init.d/S02splash:57]: -lt 0

File: board/recalbox/fsoverlay/etc/init.d/S02splash [hunk 4]
```diff
@@ -51,7 +51,15 @@ do_start ()
...
+    do
+	sleep 1
+	let N++
```
Comment (digitalLumberjack) [inline board/recalbox/fsoverlay/etc/init.d/S02splash:60]: N--

File: board/recalbox/fsoverlay/etc/init.d/S02splash (hunk 2 above)
Comment (digitalLumberjack) [inline board/recalbox/fsoverlay/etc/init.d/S02splash:56]: So we can visualize the number of try

File: board/recalbox/fsoverlay/etc/init.d/S02splash (hunk 4 above)
Comment (digitalLumberjack) [inline board/recalbox/fsoverlay/etc/init.d/S02splash]: recalbox-part.sh replace the fstab, need to change the sed in https://github.com/recalbox/recalbox-rescue/blob/rb-4/sdcontent/os/recalboxOS-rpi3/partition_setup.sh before merging (i'll do)

**Comment by digitalLumberjack** [comment]
recalbox-part.sh replace the fstab, need to change the sed in https://github.com/recalbox/recalbox-rescue/blob/rb-4/sdcontent/os/recalboxOS-rpi3/partition_setup.sh before merging (i'll do)

File: board/recalbox/fsoverlay/recalbox/scripts/recalbox-config.sh [hunk 5]
```diff
@@ -251,21 +251,33 @@ fi
 fi
 
 if [ "$command" == "audio" ];then
+    # this code is specific to the rpi
+    # don't set it on other boards
+    # find a more generic way would be nice
```
Comment (digitalLumberjack) [inline board/recalbox/fsoverlay/recalbox/scripts/recalbox-config.sh:256]: alsamixer ?

File: board/recalbox/recalbox-patch-target.sh [hunk 6]
```diff
@@ -22,6 +22,18 @@ ln -sf "/recalbox/share/cheats"                                       "${TARGET_
...
 rm -f "${TARGET_DIR}/etc/init.d/S50kodi" || exit 1
 
+# acpid requires /var/run, so, requires S03populate
+if test -e "${TARGET_DIR}/etc/init.d/S02acpid"
+then
+    mv "${TARGET_DIR}/etc/init.d/S02acpid" "${TARGET_DIR}/etc/init.d/S05acpid" || exit 1
+fi
+
+# we don't want default xorg files
+rm -f "${TARGET_DIR}/etc/X11/xorg.conf" || exit 1
```
Comment (digitalLumberjack) [inline board/recalbox/recalbox-patch-target.sh:32]: rm -f never return != 0

Comment (digitalLumberjack) [inline board/recalbox/recalbox-patch-target.sh:32]: whatever

File: package/busybox/busybox.config [hunk 7]
```diff
@@ -25,7 +25,7 @@ CONFIG_FEATURE_VERBOSE_USAGE=y
 # CONFIG_FEATURE_COMPRESS_USAGE is not set
 CONFIG_FEATURE_INSTALLER=y
 # CONFIG_INSTALL_NO_USR is not set
-CONFIG_LOCALE_SUPPORT=y
```
Comment (digitalLumberjack) [inline package/busybox/busybox.config:28]: We need that for special chars in bash no ?

**Comment by digitalLumberjack** [comment]
Review finished. Great job.
We discuss about the fex things i commented, we change the recalbox-rescue scripts and go for merge :)

**Comment by nadenislamarre** [comment]
-rm -f /tmp/nae || echo ahbon
rm: ne peut enlever `/tmp/nae': Opération non permise

## ahbon

**Comment by nadenislamarre** [comment]
> > We need that for special chars in bash no ?
> > when i did the busybox config the first time some monthes ago, i modified the bad file.
> > i just moved my busybox config to the custom file done for this.

**Comment by digitalLumberjack** [comment]
`rm -f /tmp/nae; echo $?` 
0
Seems to be shell dependent

**Comment by nadenislamarre** [comment]
no, it's an answer from rm.
note that i switch of user to get this error.

the only difference between rm and rm -f
is that
1- if the file doesn't exist, rm -f returns 0
2- no prompt

so, i case you've not the right to remove a file, both are failing for
example.

**Comment by digitalLumberjack** [comment]
okay :)

**Comment by digitalLumberjack** [comment]
Let's do that.

**Comment by digitalLumberjack** [comment]
@ian57 didn't see your message. Is it still valid ?

**Comment by digitalLumberjack** [comment]
Hum
this is what we aim : https://github.com/libretro/picodrive/blob/master/Makefile.libretro#L324

**Comment by ian57** [comment]
No your modification is ok... did not see the arm part! sorry

**Pull request merged**